### PR TITLE
Makes sure Commands/Queries handle 403 responses

### PR DESCRIPTION
**Pull request by mereghost**
## What?

This tries to make sure that all commands handle 403s correctly. 

## How?

By handling 403s! Surprising, huh? But in all seriousness there are some standardization of the way we treat the errors across different commands as well as some code cleanup where needed.

### CODE REVIEW

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb [hunk 1]
```diff
@@ -69,10 +69,12 @@ def using_user_token(storage, user, &)
                 )
             end
 
-            def storage_error(response:, code:, source:)
-              data = StorageErrorData.new(source:, payload: response.json(symbolize_keys: true))
+            def storage_error(response:, code:, source:, log_message: nil)
+              # Some errors, like timeouts, aren't json responses so we need to adapt
+              payload = response.respond_to?(:json) ? response.json(symbolize_keys: true) : response.to_s
```
Comment (Kharonus) [inline modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb:74]: 🟡 rather then using `respond_to?(:json)`, can we use `response.content_type.mime_type`? in the nextcloud util we have such a matching: 

```ruby
            def error_data_from_response(caller:, response:)
              payload =
                case response.content_type.mime_type
                when "application/json"
                  response.json
                when "text/xml", "application/xml"
                  response.xml
                else
                  response.body.to_s
                end

              StorageErrorData.new(source: caller, payload:)
            end
```

**Review by Kharonus — CHANGES_REQUESTED** [review]
There is only one issue I want to raise

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb (hunk 1 above)
Comment (mereghost) [reply modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb:74]: It doesn't seem to be the case as we have 2 things in play here: 

1. a `Response` that happens to be an error to us
2. an `HTTPXErrorResponse` which happens when the request wasn't able to be made at all

So the first one we could parse the mime_type (in theory should just be json as this is only for the MS API) but on the second one not really.

**Review by Kharonus — APPROVED** [review]

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb (hunk 1 above)
Comment (Kharonus) [reply modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb:74]: I think, the current approach is alright. In the end, it is just a subjective point of view.

**Pull request merged**